src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvokerProvider.cs

```diff
@@ -219,13 +219,10 @@ internal List<Func<IRazorPage>> GetPageStartFactories(CompiledPageActionDescript
                 PageStartFileName);
             foreach (var item in pageStartItems)
             {
-                if (item.Exists)
```

Comment:
Precompiled views don't get published so the project doesn't find these on disks. However if you ask the page factory for an instance, it happily produces one from the compiler cache. We do a similar thing with the view engine relying on the factory to produce instances regardless of file existence.